packages/abi-gen/package.json

```diff
@@ -9,38 +9,40 @@
     "types": "lib/src/index.d.ts",
     "scripts": {
         "lint": "tslint --format stylish --project . && yarn lint-contracts",
-        "lint-contracts": "solhint -c ../../contracts/.solhint.json contracts/**/**/**/**/*.sol",
+        "lint-contracts": "solhint -c ../../contracts/.solhint.json test-cli/contracts/*.sol",
         "fix": "tslint --fix --format stylish --project . && yarn lint-contracts",
-        "clean": "shx rm -rf lib test/generated-test/output",
+        "clean": "shx rm -rf lib test-cli/**/generated-wrappers",
         "build": "tsc -b",
-        "build_contracts_to_wrap": "run-s compile:sol",
-        "build_contract_wrappers": "tsc --project test/generated-test/tsconfig.json",
-        "build:ci": "yarn build",
-        "test": "run-s run_mocha build_contracts_to_wrap generate_contract_wrappers diff_contract_wrappers # build_contract_wrappers run_contract_wrapper_tests HACK: note this is commented out. building those contract wrappers is broken on development right now",
+        "test_cli:clean": "yarn clean:sol && rm -rf test-cli/**/generated-wrappers && rm -rf test-cli/typescript/lib",
+        "test_cli:prebuild": "yarn compile:sol && yarn generate_contract_wrappers",
+        "test_cli:build": "tsc --project test-cli/tsconfig.json",
+        "test_cli": "yarn test_cli:run_mocha && yarn diff_contract_wrappers",
+        "build:ci": "yarn build && yarn test_cli:prebuild && yarn test_cli:build",
+        "test": "yarn run_mocha",
         "run_mocha": "mocha --require source-map-support/register --require make-promises-safe lib/test/*_test.js --timeout 100000 --bail --exit",
-        "run_contract_wrapper_tests": "mocha --require source-map-support/register --require make-promises-safe test/generated-test/output/TypeScript/lib/**/*_test.js --timeout 100000 --bail --exit",
-        "test:circleci": "yarn test:coverage",
+        "test_cli:run_mocha": "mocha --require source-map-support/register --require make-promises-safe test-cli/typescript/lib/**/*_test.js --timeout 100000 --bail --exit",
+        "test:circleci": "yarn test:coverage && yarn test_cli",
```

Comment:
why not also integrate with `yarn test`? this looks like a developer trap: someone can easily forget to run `yarn test_cli` locally, and only find out there are problems when CI runs.

Reply:
Could do that. Should we also integrate `test_cli:build` into `yarn build`?

Reply:
Yup.